### ISSUE TITLE
[Enhancement]:  aws_ami data source doesn't support kms_key_id attribute on EBS

### ISSUE THREAD

**Issue by clayvan** [hashicorp/terraform-provider-aws#29027]
### Description

The [aws_ami data source](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/ami#ebs) should support the attribute `kms_key_id` on EBS volumes. The [Launch template block_device_mapping](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/ami#block_device_mappings) for EBS supports `kms_key_id` as an argument, and we would like to be able to use the same CMK from another account.

### Affected Resource(s) and/or Data Source(s)

aws_ami data source

### Potential Terraform Configuration

```terraform
resource "aws_launch_template" "custom_nodes" {
  ...
  block_device_mappings {
    ebs {
      ...
      kms_key_id = data.aws_ami.custom_node.block_device_mappings[0].ebs[0].kms_key_id
    }
  }
}
```

### Would you like to implement a fix?

None

**Comment by mattburgess** [comment]
Hi @clayvan. It looks like KmsKeyId isn't supported for AMIs. From https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_EbsBlockDevice.html:

> This parameter is only supported on BlockDeviceMapping objects called by [RunInstances](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_RunInstances.html), [RequestSpotFleet](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_RequestSpotFleet.html), and [RequestSpotInstances](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_RequestSpotInstances.html).

As AMIs are created using the [RegisterImage](https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_RegisterImage.html) API call then it's not supported. This is what happens when I try to specify it in a locally modified version of the Terraform AWS Provider:

```
Error: creating EC2 AMI (tf-acc-test-5616169358532386451): UnknownParameter: The parameter KmsKeyId is not recognized
        	status code: 400, request id: 29b18041-0361-4772-8790-790e2f6b9df2
```

**Comment by clayvan** [comment]
Hmm that's confusing, the [aws_ami resource type](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/ami#argument-reference) takes `kms_key_id` as an argument on the EBS volume?

Note that if you create an AMI and only pass `Encrypted` [but no KMS key ID, it will use the default of your account](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/AMIEncryption.html), so maybe it doesn't show up either when you run the `describe` on the resource again?

**Comment by mattburgess** [comment]
I got a little confused by that myself yesterday. Although it's mentioned in the docs, it isn't actually handled by the code at all: https://github.com/hashicorp/terraform-provider-aws/blob/main/internal/service/ec2/ec2_ami.go#L88-L146

When I tried adding support there that's what caused me to hit that API error. I've raised a PR to remove that parameter from the docs to hopefully prevent this confusion in the future!
